packages/suite/src/components/suite/ChangeDeviceLabel.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import { analytics, EventType } from '@trezor/suite-analytics';
+import styled, { css } from 'styled-components';
+
+import { Button, INPUT_HEIGHTS, Input } from '@trezor/components';
+import { Translation } from 'src/components/suite';
+import { useDevice, useDispatch, useTranslation } from 'src/hooks/suite';
+import * as deviceSettingsActions from 'src/actions/settings/deviceSettingsActions';
+import { MAX_LABEL_LENGTH } from 'src/constants/suite/device';
+import { SCREEN_SIZE } from '@trezor/components/src/config/variables';
+import { isAscii } from '@trezor/utils';
+
+const StyledInput = styled(Input)<{ isVertical?: boolean }>`
+    ${props =>
+        props.isVertical &&
+        css`
+            margin: 4px 0 4px 4px;
+
+            &:not(:first-child) {
+                margin-left: 8px;
+            }
+
+            @media (max-width: ${SCREEN_SIZE.SM}) {
+                margin: 0px 0 10px;
+                width: 100%;
+            }
+        `}
+`;
+
+const StyledButton = styled(Button)<{ isVertical?: boolean; isDisabled: boolean }>`
+    height: ${INPUT_HEIGHTS.large}px;
+
+    ${props =>
+        props.isVertical &&
+        css`
+            min-width: 170px;
+            margin: 4px 0 4px 4px;
+            cursor: ${props.isDisabled ? 'not-allowed' : 'pointer'};
+            &:not(:first-child) {
+                margin-left: 8px;
+
+                @media (max-width: ${SCREEN_SIZE.SM}) {
+                    margin-left: 0;
+                }
+            }
+
+            @media (max-width: ${SCREEN_SIZE.SM}) {
+                width: 100%;
+                margin: 0;
+            }
+        `}
+`;
+
+interface ChangeDeviceLabelProps {
+    isDeviceLocked: boolean;
```

Comment:
Also, this could be checked inside the component.

Reply:
Yes but then we would need to refactor every component in `SettingsDevice` so it's same for every component but it makes since there so I would leave this one as it is.

Reply:
Not necessarily. The other child component use it directly, while this one is nested one level deeper. And also reused.